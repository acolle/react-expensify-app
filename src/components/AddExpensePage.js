import React from 'react';
import { connect } from 'react-redux'; // Connect this component to the store so it can dispatch
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/'); // Redirect to the dashboard after adding an expense
  };

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

// First parenthesis is empty as we do not need the state of the store
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
