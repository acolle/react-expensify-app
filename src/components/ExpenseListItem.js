import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <div className="ExpenseListItem">
        <Link to={`/edit/${id}`}>
          <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>
      </div>
    </div>
  );
}

// To get access to dispatch from the props
export default ExpenseListItem;
