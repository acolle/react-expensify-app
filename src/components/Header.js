import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify App</h1>
    <ul>
      <li><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
      <li><NavLink to="/create" activeClassName="is-active">Add expense</NavLink></li>
      <li><NavLink to="/help" activeClassName="is-active">Help</NavLink></li>
      <button onClick={startLogout}>Logout</button>
    </ul>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
