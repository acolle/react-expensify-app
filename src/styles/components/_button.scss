.button {
  background: $blue;
  border: none;
  border-bottom: .6rem solid darken($blue, 10%);
  color: $off-white;
  display: inline-block;
  font-weight: 300;
  font-size: $l-font-size;
  line-height: 1;
  text-decoration: none;
  padding: $s-size;
}

.button--link {
  background: none;
  border: none;
  padding: 0;
}

.button--secondary {
  background: #888888;
  border: none;
  border-bottom: .6rem solid darken(#888888, 10%);
}

.button:disabled {
  opacity: .5;
}

// .button {
//   background: $purple;
//   border: none;
//   border-bottom: .3rem solid darken($purple, 10%);
//   color: white;
//   font-weight: 500;
//   padding: $s-size;
// }
//
// .button--link {
//   background: none;
//   border: none;
//   color: $off-white;
//   padding: 0;
// }
//
// @media (min-width: $desktop-breakpoint) {
//   .big-button {
//     margin-bottom: $xl-size;
//   }
// }
