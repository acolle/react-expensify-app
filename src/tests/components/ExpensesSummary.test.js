import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={85}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={45} expensesTotal={234765453}/>);
  expect(wrapper).toMatchSnapshot();
});
