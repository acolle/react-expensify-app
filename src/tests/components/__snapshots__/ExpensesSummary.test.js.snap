// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $0.85
  </p>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    45
     
    expenses
     totalling 
    $2,347,654.53
  </p>
</div>
`;
